extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +131,224 @@ export class Transport {
+        transportOptions['extraData']);
+      ampDoc.body.appendChild(frame);
+    }
+    // Regardless of whether we just created it, or are re-using an existing
+    // one, send extra data if available, and wire up the response callback
+    if (processResponse) {
+      const imc = Transport.crossDomainIframes_[frameUrl].iframeMessagingClient;
+      imc.registerCallback(
+        'ampAnalyticsResponse', msg => {
+        if (msg && msg.ampAnalyticsResponse) {
+          processResponse(msg.ampAnalyticsResponse);
+        }
+      });
+    }
+  }
+
+  /**
+   * Called when a creative no longer needs its cross-domain iframe (for
+   * instance, because the creative has been removed from the DOM).
+   * Once all creatives using a frame are done with it, the frame can be
+   * destroyed.
+   * @param {!HTMLDocument} ampDoc The AMP document
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   */
+  doneWithCrossDomainIframe(ampDoc, transportOptions) {
+    const frameUrl = transportOptions['iframe'];
+    if (Transport.hasCrossDomainFrame(frameUrl) &&
+      Transport.decrementFrameUsageCount_(frameUrl) <= 0) {
+      ampDoc.body.removeChild(Transport.crossDomainIframes_[frameUrl].frame);
+      delete Transport.crossDomainIframes_[frameUrl];
+    }
+  }
+
+  /**
+   * Returns whether a url of a cross-domain frame is already known
+   * @param {!string} frameUrl
+   * @return {!boolean}
+   */
+  static hasCrossDomainFrame(frameUrl) {
+    return hasOwn(Transport.crossDomainIframes_, frameUrl);
+  }
+
+  /**
+   * Sends extra data (from the amp-analytics config object) to the
+   * cross-domain iframe.
+   * @param {!string} frameUrl The URL of the frame to send the data to
+   * @param {?string=} extraData The data to send to the frame
+   */
+  sendExtraData(frameUrl, extraData) {
+    if (extraData) {
+      const frameData = Transport.crossDomainIframes_[frameUrl];
+      if (!frameData.isReady) {
+        timerFor(window).delay(() => {
+          this.sendExtraData(frameUrl, extraData);
+        }, 10);
+        return;
+      }
+      frameData.iframeMessagingClient.sendMessage('ampAnalyticsExtraData',
+        {senderId: this.id_, ampAnalyticsExtraData: extraData});
+    }
+  }
+  /**
+   * Create a cross-domain iframe for third-party vendor anaytlics
+   * @param {!HTMLDocument} ampDoc  The document node of the parent page
+   * @param {!string} frameUrl  The URL of the cross-domain iframe
+   * @return {!Element}
+   */
+  createCrossDomainIframe_(ampDoc, frameUrl, extraData) {
+    // DO NOT MERGE THIS
+    // Warning: the scriptSrc URL below is only temporary. Don't merge
+    // before getting resolution on that.
+    const frame = createElementWithAttributes(ampDoc, 'iframe', {
+      sandbox: 'allow-scripts',
+      name: JSON.stringify({
+        'scriptSrc': '/examples/analytics-3p-remote-frame-helper.js',
```

Comment:
As mentioned in the comments of /examples/analytics-3p-remote-frame-helper.js, upon approval that frame will be placed at a public/static URL. Once that is done, the value on line 249 must change to that URL.